cpmpy/solvers/minizinc.py

```diff
@@ -223,90 +272,19 @@ def objective_value(self):
         """
         return self.objective_value_
 
-    def __add__(self, cons):
-        """
-            Add an additional (list of) constraints to the model
-        """
-        if not is_any_list(cons):
-            cons = [cons]
+    def solver_var(self, cpm_var):
+        pass
 
-        txt_cons = ""
-        for con in cons:
-            txt_cons += "constraint {};\n".format(self.convert_expression(con))
+    def _post_constraint(self, cpm_expr):
 
+        txt_cons = f"constraint {self._convert_expression(cpm_expr)};\n"
         self.mzn_model.add_string(txt_cons)
-        return self
 
-    def minimize(self, expr):
-        """
-            Minimize the given objective function
-
-            `minimize()` can be called multiple times, only the last one is stored
-        """
-        # do not add it to the model, support only one 'solve' entry
-        self.mzn_txt_solve = "solve minimize {};\n".format(self.convert_expression(expr))
-
-    def maximize(self, expr):
-        """
-            Maximize the given objective function
-
-            `maximize()` can be called multiple times, only the last one is stored
-        """
-        # do not add it to the model, support only one 'solve' entry
-        self.mzn_txt_solve = "solve maximize {};\n".format(self.convert_expression(expr))
 
     def clean_varname(self, varname):
```

Comment:
We had a private discussion on solver_var for minizinc.

I think, to keep everything unified, solver_var() should maintain which variables are already created, and if not yet, to add them to the mzn file on the fly.
Then, you can move this 'clean_varname' into solver_var: that will be the only place where you will need it.
And then, solver_var(cpm_var) actually returns the solver_var: the minizinc-compatible string name of the variable?

then the 'add variables' part can go out of __add__ just like it is not in the other interfaces
and the 'fill variables' will also become similar: for all vars in user_vars, call solver_var() to get their true handle, and get the corresponding value?


So if you ca

Reply:
Yes I agree that everything should go through solver_var(). Indeed, if we just return the minizinc-compatible string, a user can do the following once he has a solution: `mzn_var = cpm_var; sol.mzn_var`, just like in other solvers.
Still a little bit weird as we don't "post" a new variable to the minizinc interface, but this way at least the interface is unified.

On another note: maybe it would be good to create some sample scenarios how to use CPMpy and just swap out different solvers, so to make sure that all interfaces are unified?